### PR TITLE
refactor(container.provider.test): refactor ContainerInstance unit tests

### PR DESCRIPTION
**Pull request by mattdibi**
Refactoring of `ContainerInstance` unit tests required by #5129

### CODE REVIEW

File: kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java [hunk 1]
```diff
@@ -95,6 +101,18 @@ public void deactivate() {
         logger.info("deactivate...done");
     }
 
+    public synchronized ContainerInstanceState getState() {
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java]: What about returning just `this.state.getClass().getSimpleName()`? In this way adding or removing a state does not require keeping the enum `ContainerInstanceState` up to date. Something similar is done in log at line 140.

Comment (mattdibi) [reply kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java]: I don't like passing around the `State` reference. What if a user attempts to change it/call a method that it shouldn't call? This was my thinking... passing around only the name could be a good idea though...

Comment (marcellorinaldo) [reply kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java]: Yes, I was referring to just the name

Comment (mattdibi) [reply kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java]: Addressed in 4bd4975eb71b45ac01da4ff0b1c42350b2b67339

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**